File: singleNumber.js
// https://leetcode.com/problems/single-number/

const singleNumber = function(nums) {
	let result = 0;
	for (let i = 0; i < nums.length; i++) {
		result ^= nums[i];
	}
	return result;
};

const test1 = [1, 2, 3, 3, 2];
const test2 = [2, 3, 4, 5, 6, 4, 3, 2, 1, 5, 1];
console.assert(singleNumber(test1) === 1, "works for test-case 1");
console.assert(singleNumber(test2) === 6, "works for test-case 2");

File: singleNumber.py
# https://leetcode.com/problems/single-number/submissions/

class Solution:
	def singleNumber(self, nums) -> int:
		res = 0
		for i in nums:
			res ^= i
		return res

test = Solution()

print(test.singleNumber([4, 1, 2, 1, 2]), 4)
